packages/@o3r/logger/src/services/logger/logger.spec.ts

```diff
@@ -102,60 +121,81 @@ describe('Logger service', () => {
     it('should stop and resume recording', () => {
       service.stopClientRecording();
 
-      expect(client.stopRecording).toHaveBeenCalled();
+      expect(stopRecordingMock).toHaveBeenCalled();
 
       service.resumeClientRecording();
 
-      expect(client.resumeRecording).toHaveBeenCalled();
+      expect(resumeRecordingMock).toHaveBeenCalled();
     });
 
     it('should report logs', () => {
       service.log('log', { foo: 'bar' });
 
-      expect(client.log).toHaveBeenCalledWith('log', { foo: 'bar' });
+      expect(logMock).toHaveBeenCalledWith('log', { foo: 'bar' });
 
       service.warn('warn', { foo: 'bar' });
 
-      expect(client.warn).toHaveBeenCalledWith('warn', { foo: 'bar' });
+      expect(warnMock).toHaveBeenCalledWith('warn', { foo: 'bar' });
 
       service.error('error', { foo: 'bar' });
 
-      expect(client.error).toHaveBeenCalledWith('error', { foo: 'bar' });
+      expect(errorMock).toHaveBeenCalledWith('error', { foo: 'bar' });
     });
 
     it('should create a meta reducer', () => {
       service.createMetaReducer();
 
-      expect(client.createMetaReducer).toHaveBeenCalled();
+      expect(createMetaReducerMock).toHaveBeenCalled();
     });
   });
 
   describe('with multiple third-party client', () => {
     let service: LoggerService;
     let client1: LoggerClient;
     let client2: LoggerClient;
+    const stopRecordingMock1 = jest.fn();
+    const resumeRecordingMock1 = jest.fn();
+    const stopRecordingMock2 = jest.fn();
+    const resumeRecordingMock2 = jest.fn();
+    const errorMock1 = jest.fn();
+    const warnMock1 = jest.fn();
+    const logMock1 = jest.fn();
+    const errorMock2 = jest.fn();
+    const warnMock2 = jest.fn();
+    const logMock2 = jest.fn();
+
     describe('via forRoot', () => {
       beforeEach(() => {
+        stopRecordingMock1.mockReset();
+        resumeRecordingMock1.mockReset();
+        stopRecordingMock2.mockReset();
+        resumeRecordingMock2.mockReset();
+        errorMock1.mockReset();
+        warnMock1.mockReset();
+        logMock1.mockReset();
+        errorMock2.mockReset();
+        warnMock2.mockReset();
+        logMock2.mockReset();
         client1 = {
           identify: jest.fn(),
           event: jest.fn(),
           getSessionURL: jest.fn().mockReturnValue('client.session.url1'),
-          stopRecording: jest.fn(),
-          resumeRecording: jest.fn(),
-          error: jest.fn(),
-          warn: jest.fn(),
-          log: jest.fn(),
+          stopRecording: stopRecordingMock1,
```

Comment:
Is it coming from a Rule?
This change is fine but a mockReset instead of a new instance can lead to issue if not correctly applied

Reply:
Most of the time it's to fix [@typescript-eslint/unbound-method](https://typescript-eslint.io/rules/unbound-method/) 
I run the test in local for this package and they are still green :)

Reply:
I knew the tests were fine.
The `jest.fn()` is not part of the exceptions handled by `eslint-plugin-jest` (as described in the linked doc)?

Reply:
issue is not on `jest.fn()` but on `expect(client.log)` because we gave the reference to an unbound method

Reply:
I got your point 
I was not aware this rule [jest/unbound-method](https://github.com/jest-community/eslint-plugin-jest/blob/HEAD/docs/rules/unbound-method.md)
This rule is not activated on our config 
we may check if we want to activate it or not

Reply:
We will see later if we can/should use [jest/unbound-method](https://github.com/jest-community/eslint-plugin-jest/blob/HEAD/docs/rules/unbound-method.md)